### PR TITLE
pause: Remove everything related to pause binary

### LINKED ISSUE #755: Remove everything related to pause container
# Description of problem
We needed a pause binary in order to run a pause container because our former agent could not handle a pod running once all containers were removed.
Our current agent does not need such a workaround.

### PR DESCRIPTION
**Pull request by sboeuf**
Now that virtcontainers has removed the pause binary because there is no need for using a pause container anymore, the runtime repository does no longer need to have it either.
This PR vendors virtcontainers and remove all references to the pause binary.

Fixes #755

### CODE REVIEW

**Comment by gorozco1** [comment]
We need a PR in https://github.com/clearcontainers/packaging to remove pause as well after this PR is merged.

**Comment by sboeuf** [comment]
I set this as do-not-merge since we need the PR on virtcontainers to be merged before I can properly revendor virtcontainers.
This PR was about testing the whole thing was working.

File: cc-env.go [hunk 1]
```diff
@@ -97,9 +97,8 @@ type ShimInfo struct {
 
 // AgentInfo stores agent details
 type AgentInfo struct {
-	Type         string
-	Version      string
-	PauseBinPath string
+	Type    string
```
Comment (jodh-intel) [inline cc-env.go]: As you are changing the output of `cc-env`, please can you bump `formatVersion` in `cc-env.go`.

Comment (sboeuf) [reply cc-env.go]: Done

**Comment by sameo** [comment]
LGTM

**Comment by sboeuf** [comment]
@iphutch can you give us the approval here please ? This basically removes documentation, nothing has been added.

**Comment by sboeuf** [comment]
thx @iphutch !

**Pull request merged**